flink-kubernetes/src/main/java/org/apache/flink/kubernetes/KubernetesResourceManager.java

```diff
@@ -272,10 +267,9 @@ private void requestKubernetesPod() {
 	 * Request new pod if pending pods cannot satisfy pending slot requests.
 	 */
 	private void requestKubernetesPodIfRequired() {
-		final int requiredTaskManagerSlots = getNumberRequiredTaskManagerSlots();
-		final int pendingTaskManagerSlots = numPendingPodRequests * numSlotsPerTaskManager;
+		final int requiredTaskManagers = getNumberRequiredTaskManagers();
 
-		if (requiredTaskManagerSlots > pendingTaskManagerSlots) {
+		if (requiredTaskManagers > numPendingPodRequests) {
```

Comment:
I can see the return value of `startWorker` is `@return whether the resource can be allocated`

If so, we should also return boolean in this method where false if we don't request a new pod.

Reply:
`startNewWorker` is a public interface, and its return value is checked by `SlotManager` for creating pending slots.
For `requestKubernetesPodIfRequired`, I don't see how its return value is needed, at least for now. And since it's a internal private method, I think we can always add that return value if needed later.

Reply:
Oops I mix up `requestKubernetesPod` with `requestKubernetesPodIfRequired`.